py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/s3_utils.py

```diff
@@ -26,13 +31,19 @@ def download_file_from_s3(bucket_name: str,
     """
     simple wrapper to download a file from s3 to a directory.
     """
-    print("downloading %s" % filename)
-    s3 = boto3.resource('s3')
-
-    # have to convert filenames, deliminating with . instead of /, or else the
-    # move method in the download_file method errors out with "no directory".
-    deliminated_filename = filename.replace("/", ".") 
-
-    destination = os.path.join(destination_dir, deliminated_filename)
-    s3.Bucket(bucket_name).download_file(filename, destination)
-    return Path(destination)
+    logging.info(
+        "Downloading %s from s3 bucket %s to %s" % (filename,
+                                                    bucket_name,
+                                                    destination_dir))
+    try:
+        s3 = boto3.resource('s3')
+
+        # have to convert filenames, deliminating with . instead of /, or else the
+        # move method in the download_file method errors out with "no directory".
+        deliminated_filename = filename.replace("/", ".")
+
+        destination = os.path.join(destination_dir, deliminated_filename)
+        s3.Bucket(bucket_name).download_file(filename, destination)
+        return Path(destination)
+    except Exception as exception:
+        raise S3Exception("Failed to download %s" % filename) from exception
```

Comment:
decided to re-raise here.

if we're not downloading things for some reason, i think we'll want our lambda jobs to fail.

Reply:
I'm not sure we would want to raise here, as it would cause the whole batch to fail because of an error on one file. 

To keep the batching approach as fault-tolerant as possible, it would probably be advantageous to have a structure where errors caused by singular files have little to no impact on the batch as a whole.